devdoc/buffer_requirements.md

```diff
@@ -49,7 +52,26 @@ BUFFER_create creates a new buffer from the memory at source, having size "size"
 
 **SRS_BUFFER_02_004: [** Otherwise, BUFFER_create shall return a non-NULL handle. **]**
 
+### BUFFER_create_with_size
+
+
+
+```c
+extern BUFFER_HANDLE BUFFER_create_with_size(size_t buff_size)
+```
+
+**SRS_BUFFER_07_029: [** BUFFER_create_with_size shall create a BUFFER_HANDLE with a pre allocated underlying buffer size. **]**
+
+**SRS_BUFFER_07_030: [** If buff_size is 0 BUFFER_create_with_size shall return NULL. **]**
```

Comment:
>buff_size is 0 BUFFER_create_with_size shall return NULL. [](start = 28, length = 57)

This should not return NULL. It should be possible to allocate a buffer with size 0.

For the same reason that malloc allows it. Here goes.

During writing SW, sometimes one would write stuff like:

size_t payload_size = get_my_payload_size(); // for example read it from a packet on the wire

BUFFER_HANDLE payload_buffer = BUFFER_create_with_size(payload_size);

Notice how the code does not need to differentiate between a zero size and non zero size payload. That is neat because why should we litter the callers of malloc/BUFFER_create_with_size with 0 checks? 0 or positive payload size are both valid cases.

Can you please address this? otherwise I fear that assumptions can be easily made that BUFFER_create_with_size will work with 0 and it could potentially introduce bugs down the road.

And now it is the time to change it as it is very cheap (we're in the code :-)).